### PR TITLE
create a schedule with a list of courses

### PR DESCRIPTION
**Pull request by GianlucaP106**

### CODE REVIEW

**Comment by mattcab2002** [comment]
Idea for this is the frontend would handle the OCR then send a POST request to the create-schedule endpoint with all the classes the OCR detected?

File: mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/TO/schedule/CourseTO.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.util.Date;
+import java.util.List;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
```
Comment (mattcab2002) [inline mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/TO/schedule/CourseTO.java:11]: genius annotations

**Comment by abhijeetpraveen** [comment]
> Idea for this is the frontend would handle the OCR then send a POST request to the create-schedule endpoint with all the classes the OCR detected?

I have opened an issue #64 for handling the OCR from the backend. I will work on that ticket and implement OCR with Java to create a Course List from a schedule that we can display in the frontend.

**Comment by mattcab2002** [comment]
So upon upload of a schedule you would perform the OCR?

**Comment by mattcab2002** [comment]
Before saving it to the DB^

**Comment by GianlucaP106** [comment]
I created the endpoint for completion but we can just use the service method with the list of courses, if the OCR is done on the backend we can change the endpoint to take in the file and just call the service method

**Comment by abhijeetpraveen** [comment]
> Before saving it to the DB^

We would save it to the DB and do the OCR as well. The schedule can still be displayed as a file on the frontend so that people can look at it. So we'll save it to the db and then using the services that have already been defined by Gig, I will implement the OCR to work with them.

**Comment by abhijeetpraveen** [comment]
> I created the endpoint for completion but we can just use the service method with the list of courses, if the OCR is done on the backend we can change the endpoint to take in the file and just call the service method

we can implement the OCR on the upload Schedule endpoint, and then from there we can call the service methods to create a weekly schedule and the courses from that schedule.

**Comment by mattcab2002** [comment]
> > Before saving it to the DB^
> 
> 
> 
> We would save it to the DB and do the OCR as well. The schedule can still be displayed as a file on the frontend so that people can look at it. So we'll save it to the db and then using the services that have already been defined by Gig, I will implement the OCR to work with them. 


I was thinking we would make the OCR check before saving to DB. It would prevent us from saving an unnecessary image and it would provide us a way to handle a "failed" upload on the frontend by returning a 403 or some other status code. 

On the return from the upload it would be ideal to return the course codes so we could use it on the frontend to display the courses we found

**Comment by abhijeetpraveen** [comment]
> > > Before saving it to the DB^
> > 
> > 
> > 
> > We would save it to the DB and do the OCR as well. The schedule can still be displayed as a file on the frontend so that people can look at it. So we'll save it to the db and then using the services that have already been defined by Gig, I will implement the OCR to work with them. 
> 
> 
> I was thinking we would make the OCR check before saving to DB. It would prevent us from saving an unnecessary image and it would provide us a way to handle a "failed" upload on the frontend by returning a 403 or some other status code. 
> 
> On the return from the upload it would be ideal to return the course codes so we could use it on the frontend to display the courses we found

Yes let's do it this way. It makes more sense. I'll implement it this way.

**Comment by GianlucaP106** [comment]
Ok so review?
lol

File: mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/services/schedule/WeeklyScheduleService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        schedule = weeklyScheduleRepository.save(schedule);
+        userInfo.setSchedule(schedule);
+        appUserInfoService.save(userInfo);
+        for (CourseTO course : courses) {
+            courseService.createCourse(course, schedule);
+        }
+        return schedule;
+    }
+
+    public WeeklyScheduleTO from(WeeklySchedule schedule) {
```
Comment (abhijeetpraveen) [inline mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/services/schedule/WeeklyScheduleService.java:40]: Can we name the method something else, from is a bit vague no?

Comment (mattcab2002) [reply mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/services/schedule/WeeklyScheduleService.java:40]: I think its fine its just the TO

**Review by mattcab2002 — APPROVED** [review]
Good with me

**Review by abhijeetpraveen — APPROVED** [review]
LGTM

**Pull request merged**